File: src/main/java/com/dh/ondot/member/infra/RedisTokenRepository.java
package com.dh.ondot.member.infra;

import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Component;

import java.time.Duration;

@Component
@RequiredArgsConstructor
public class RedisTokenRepository {

    private static final String BLACKLIST_KEY_PREFIX = "blacklist:";

    private final RedisTemplate<String, String> redisTemplate;

    public boolean isBlacklisted(String jti) {
        return redisTemplate.opsForValue().get(toBlacklistKey(jti)) != null;
    }

    public void addBlacklistToken(String jti, Duration expiration) {
        redisTemplate.opsForValue().set(toBlacklistKey(jti), jti, expiration);
    }

    private String toBlacklistKey(String jti) {
        return BLACKLIST_KEY_PREFIX + jti;
    }
}

File: src/main/kotlin/com/dh/ondot/member/infra/RedisTokenRepository.kt
package com.dh.ondot.member.infra

import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Component
import java.time.Duration

@Component
class RedisTokenRepository(
    private val redisTemplate: RedisTemplate<String, String>,
) {
    fun isBlacklisted(jti: String): Boolean {
        return redisTemplate.opsForValue().get(toBlacklistKey(jti)) != null
    }

    fun addBlacklistToken(jti: String, expiration: Duration) {
        redisTemplate.opsForValue().set(toBlacklistKey(jti), jti, expiration)
    }

    private fun toBlacklistKey(jti: String): String {
        return BLACKLIST_KEY_PREFIX + jti
    }

    companion object {
        private const val BLACKLIST_KEY_PREFIX = "blacklist:"
    }
}
